misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/BallerinaStackFrame.java

```diff
@@ -128,13 +135,41 @@ private static String getStackFrameName(StackFrameProxyImpl stackFrame) {
                 return stackFrame.visibleVariableByName(STRAND_VAR_NAME) != null ? frameName :
                         FRAME_TYPE_START + FRAME_SEPARATOR + frameName;
             } else {
-                return stackFrame.location().method().name();
+                return getFilteredStackFrame(stackFrame);
             }
         } catch (Exception e) {
             return FRAME_TYPE_ANONYMOUS;
         }
     }
 
+    /**
+     * Derives filtered ballerina stack frame name from the given java stack frame instance.
+     *
+     * @param stackFrame JDI stack frame instance
+     * @return filtered ballerina stack frame name
+     */
+    private static String getFilteredStackFrame(StackFrameProxyImpl stackFrame) throws JdiProxyException {
+        String stackFrameName = stackFrame.location().method().name();
+        LocalVariableProxyImpl selfVisibleVariable = stackFrame.visibleVariableByName(SELF_VAR_NAME);
+        if (selfVisibleVariable == null) {
+            return stackFrameName;
+        }
+        Value stackFrameValue = stackFrame.getValue(selfVisibleVariable);
+        if (isService(stackFrameValue)) {
+            if (stackFrameName.equals(METHOD_INIT)) {
+                return BVariableType.SERVICE.getString();
+            } else {
+                // Here stackFrameName format will be `$resourceAccessor$resourceName$otherResourceParts`
+                String[] stackFrameNameParts = stackFrameName.split("\\$");
+                if (!stackFrameNameParts[1].isEmpty() && stackFrameNameParts[1].equals(ACCESSOR_DEFAULT)) {
```

Comment:
Please check size of the `stackFrameNameParts` before accessing its elements, to avoid  `indexoutofboundsexception`

Reply:
What does this `ACCESSOR_DEFAULT` mean? `stackFrameNameParts[1]` is not the resource accessor but the resource name as per your comment, right?

Reply:
We can use `default` for the accessor. Refer:
https://ballerina.io/learn/by-example/http-default-resource

```
stackFrameNameParts[0] = "";
stackFrameNameParts[1] = resourceAccessor; // (default, get, post, pot, delete)
stackFrameNameParts[2] = resourcePath/resourceName/
```

Reply:
I see.. Thanks for the clarification